packages/dds/tree/api-report/tree.public.api.md

```diff
@@ -123,6 +123,26 @@ export type InsertableTypedNodeUnsafe<T extends Unenforced<TreeNodeSchema>> = Un
 export interface InternalTreeNode extends ErasedType<"@fluidframework/tree.InternalTreeNode"> {
 }
 
+declare namespace InternalTypes {
```

Comment:
One of the main goals of the private scopes for types is to make things less public.

Here we can see this change is adding a bunch of stuff to the public API that was previously not in it.

I don't think its worth the effort to migrate types that are already `@internal` into internal scopes, and we certainly don't want to cause them to show up in the public API if they weren't already in it.

I think what we are seeing is a limitation of API extractor, and it not correctly filtering internal items in namespaces in the public API report.

Of more significant is the generated public.d.ts file which we produce when compiling tree. It seems to have the same limitation from what I can tell, and also doesn't get filtering.

Given these limitations I think the best way forward is to just worry about `@public` APIs for now, and just create a single `internalTypes` module at the top level, and only put types which are logically internal implementations details of `@public` types in it.

This accomplishes a few things:

- It gives us only as single internal scope to deal with
- Avoids leaking concepts like FlexList in the name of the top levels exports, which itself is an implementation detail
- Leaves all the non-public stuff alone. Its much lower priority and can be adjusted later if we decide we want to.
- Avoids the issue with API-extractor reports for namespace members with different tags
- Avoids the issue with our rollup generator not supporting namespaces fully.

Reply:
Grouping the existing internalTypes exports into a single entry point has been done in https://github.com/microsoft/FluidFramework/pull/21456